exercises/practice/queen-attack/libs/exercism/test

```diff
@@ -0,0 +1,182 @@
+    test-count @ n:=
+;
+
+\ returns true if x is a date, false otherwise
+: date? \ x -- x T
+    dup >kind ns:d n:=
+;
+
+\ adapted from 8th forum -- https://8th-dev.com/forum/index.php/topic,2745.0.html
+: eq? \ x x -- T
+   \ are the items the same kind?
+   2dup >kind swap >kind n:=
+   !if 2drop false ;then
+
+   \ same kind: try different comparators
+   number? if n:= ;then
+   string? if s:= ;then
+   array?  if ' eq? a:= 2nip ;then
+   map?    if ' eq? m:= 2nip ;then
+   date?   if d:= ;then
+
+   \ otherwise fall back to 'lazy evaluation'
+   G:l: =
+;
+
+: eps_eq? \ n x x -- T
+   \ are the items the same kind?
+   2dup >kind swap >kind n:=
+   !if 2drop false ;then
+   number? !if 2drop false ;then  
+   rot n:~=
+;
+
+: check-depth \ ... n -- ...
+  dup>r
+  n:1+ depth n:=
+  !if
+    con:red con:onBlack 
+    "PANIC: expected stack depth to be " . r> . cr
+    "Stack is:" . cr
+    .s cr
+    255 die
+  then
+  rdrop
+;
+
+\ -----------------------------------------------------------------
+
+\ status report at end of run
+( all-tests-run?  
+  !if con:red con:onBlack "... FAIL - not all tests completed" . con:white con:onBlack cr then
+) onexit
+
+\ Print a summary of the tests run
+( con:white con:onBlack
+  test-count   @ . space "tests planned - " .
+  tests-passed @ . space "passed - " .
+  tests-skipped @ . space "skipped - " .
+  tests-failed @ . space "failed" . cr
+) onexit
+
+\ -----------------------------------------------------------------
+\ The public-facing words
+\ -----------------------------------------------------------------
+    
+: equal? \ s x w -- | s w x --
+   run-test? !if 2drop test-skipped ;; then
+   isword? !if swap then
+   w:exec
+   3 check-depth
+   2dup             \ so test-failed can show actual and expected
+   eq? if test-passed else test-failed then
+;
+
+: approx_equal? \ s x w n -- | s w x n --
+   run-test? !if 3drop test-skipped ;; then
+   -rot isword? !if swap then
+   w:exec
+   4 check-depth
+   3dup            \ so test-failed can show actual and expected
+   eps_eq?
+   if rot drop test-passed else rot drop test-failed then
+;
+
+: true? \ s w --
+   run-test? !if drop test-skipped ;; then
+   w:exec
+   2 check-depth
+   true swap dup    \ so test-failed can show actual and expected
+   if test-passed else test-failed then 
+;
+
+: false? \ s w --
+   run-test? !if drop test-skipped ;; then
+   w:exec
+   2 check-depth
+   false swap dup    \ so test-failed can show actual and expected
+   !if test-passed else test-failed then 
+;
+
+: null? \ s w --
+   run-test? !if drop test-skipped ;; then
+   w:exec
+   2 check-depth
+   null swap dup     \ so test-failed can show actual and expected
+   G:null? nip if test-passed else test-failed then 
+;
+
+: not-null? \ s w --
+   run-test? !if drop test-skipped ;; then
+   w:exec
+   2 check-depth
+   null swap dup     \ so test-failed can show actual and expected
+   G:null? not nip if test-passed else test-failed then 
+;
```

Comment:
This looks new. Please add it to (/libs/exercism/test](https://github.com/exercism/8th/blob/cb1071b78ae9221290635b68a00cbb301ee81367/libs/exercism/test#L154)

Reply:
Yes, that was discussed here: https://github.com/exercism/8th/pull/222#discussion_r1948721660

Reply:
I want to do that in a separate PR

Reply:
Apologies, I didn't read the resolved thread.

Reply:
No problem. PR has been opened: https://github.com/exercism/8th/pull/226